core/domain/exp_services.py

```diff
@@ -3025,35 +2999,20 @@ def get_user_exploration_data(
         and exp_user_data.draft_change_list else None)
     draft_change_list_id = (
         exp_user_data.draft_change_list_id if exp_user_data else 0)
-    exploration_email_preferences = (
+
+    editor_dict = exp_domain.UserExplorationData(
+        exploration=exploration,
+        states=states,
+        rights=rights_manager.get_exploration_rights(exploration_id).to_dict()
+    ).to_dict()
+
+    if user_id is not None:
+        editor_dict['draft_changes'] = draft_changes
```

Comment:
These fields should be part of UserExplorationData, I think -- generally dict conversions should be a straight to_dict() without custom logic.

Reply:
Done.